Cron/RetrieveReviews.php

```diff
@@ -2,90 +2,108 @@
 
     public function __construct(
-        Context $context,
-        StoreManagerInterface $storeManager,
         Curl $curl,
-        Dom $converter,
+        Json $jsonSerializer,
         LoggerInterface $logger,
-        VariableFactory $variable
+        VariableFactory $variable,
+        Dom $converter,
+        Config $config
     ) {
         $this->curl = $curl;
-        $this->storeManager = $storeManager;
+        $this->jsonSerializer = $jsonSerializer;
         $this->converter = $converter;
         $this->log = $logger;
         $this->variable = $variable;
-
-        $this->filePath = $this->getFeedUrl();
+        $this->config = $config;
     }
 
+    /**
+     * @throws CronException|NoSuchEntityException
+     */
+    public function processAggregateScores(): void
+    {
 
-    public function getFeedUrl(){
-        return $this->storeManager->getStore()->getConfig(
-            'kiyoh_settings/general/feed_url'
-        );
-    }
-
-    public function processAggregateScores() {
+        if (!$this->config->getLocationId()) {
+            throw new CronException(__('Location ID missing, please set your location ID.'));
+        }
 
-        $this->curl->get($this->filePath);
-        $output = $this->curl->getBody();
+        if (!$this->config->getApiKey()) {
+            throw new CronException(__('API key missing, please set your API key.'));
+        }
 
-        $doc = simplexml_load_string($output);
+        $this->curl->addHeader('X-Publication-Api-Token', $this->config->getApiKey());
+        $this->curl->get(config::PUBLICATION_URL . $this->config->getLocationId());
 
-        unset($doc->{'reviews'});
+        $output = $this->curl->getBody();
+        $jsonOutput = $this->jsonSerializer->unserialize($output);
 
         try {
-            if (isset($doc->errorCode)) {
-                throw new \Magento\Framework\Exception\CronException(__('Invalid xml format, code data missing.'));
-            }
+            if (isset($jsonOutput->errorCode)) {
 
-            $this->_saveToDb($doc);
+                throw new CronException( __(
+                    'Kiyoh API error: %1: %2',
+                    $jsonOutput->errorCode,
+                    $jsonOutput->detailedError[0]->message ?? null
+                ));
+            }
+            $this->saveToDb($jsonOutput);
 
         } catch (\Exception $e) {
             $this->log->critical($e->getMessage());
         }
     }
 
-    protected function _saveToDb($data) {
-        // save these attributes in custom var DB numberReviews averageRating
-        $ratingCodes = ['numberReviews', 'averageRating', 'percentageRecommendation'];
+    /**
+     * @param $data
+     * @return void
+     */
+    protected function saveToDb($data) : void {
+        $ratingCodes = ['numberReviews', 'averageRating', 'recommendation'];
         foreach ($ratingCodes as $ratingCode) {
-            $reviewData = $data->{$ratingCode};
+            $reviewData = $data[$ratingCode];
```

Comment:
Do we always get a value for all ratingCodes? If not, I'd fix this by doing `$reviewData = $data[$ratingCode] ?? null`

Reply:
Yes, the rating codes will be 0 if there are none, but always have the same key